nullable_types.go

```diff
@@ -0,0 +1,150 @@
+package graphql
+
+import (
+	"fmt"
+)
+
+// NullString is a string that can be null. Use it in input structs to
+// differentiate a value explicitly set to null from an omitted value.
+// When the value is defined (either null or a value) Valid is true.
+type NullString struct {
+	Value *string
+	Valid bool
+}
+
+func (NullString) ImplementsGraphQLType(name string) bool {
+	return name == "String"
+}
+
+func (s *NullString) UnmarshalGraphQL(input interface{}) error {
+	s.Valid = true
```

Comment:
I wonder if we need to leave this as `Valid`. Isn't `Set` a better choice for a name? I know that this is consistent with the `sql` package. But set sounds better, IMHO.

```go
if args.Input.Int.Valid {
}
```

vs 

```go
if args.Input.Int.Set {
}
```

Reply:
I agree, `Set` is better. I've updated the PR.